pkg/destroy/azure/azure.go

```diff
@@ -469,11 +469,11 @@ func deleteResourceGroup(ctx context.Context, client resources.GroupsClient, log
 
 	delFuture, err := client.Delete(ctx, name)
 	if err != nil {
-		if wasNotFound(delFuture.Response()) {
+		if delFuture.FutureAPI != nil {
```

Comment:
I feel like we should be checking something in `err` here rather than assuming that a nil `delFuture.FutureAPI` means that the resource group does not exist. I expect that, if there was an error, then `delFuture.FutureAPI` will be nil, regardless of whether the error was due to the resource group not existing or not.

Maybe it would be helpful to run the destroyer using a non-existent resource group to see what the value of `err` is in that case.

Reply:
@staebler you are right: `err` should have information on 404 objects. This is the pattern we use to check for not founds in [ARO](https://github.com/azure/aro-rp):

```go
if detailedErr, ok := err.(autorest.DetailedError); ok &&
	detailedErr.StatusCode == http.StatusNotFound {
	err = nil
}
if err != nil {
	return err
}
```

Reply:
Note: the above is an example from a get request. It might be enough for deleting resource groups.

We also have [this](https://github.com/Azure/ARO-RP/blob/37b20109e06813e38cbfb6f0edf433119ba8edba/pkg/util/azureerrors/error.go#L93-L108) function to check for `ResourceGroupNotFound` specificly and [this](https://github.com/Azure/ARO-RP/blob/37b20109e06813e38cbfb6f0edf433119ba8edba/pkg/cluster/delete.go#L332-L349) is how we delete cluster's resource group.

Checking for errors in Azure is hard :)

Reply:
@m1kola and @staebler Thanks for the examples and suggestions. PTAL at the new changes and let me know what you guys think.